flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.sink.serializer;
+
+import com.google.cloud.flink.bigquery.sink.exceptions.BigQuerySerializationException;
+import com.google.protobuf.ByteString;
+import org.apache.avro.generic.GenericRecord;
+
+/** Serializer for converting Avro's {@link GenericRecord} to BigQuery proto. */
+public class AvroToProtoSerializer implements BigQueryProtoSerializer<GenericRecord> {
+
+    @Override
+    public ByteString serialize(GenericRecord record) throws BigQuerySerializationException {
+        throw new UnsupportedOperationException("serialize method is not supported");
```

Comment:
since we're defining signatures, can we add the test class for this (and the other classes) too? it's fine if the only test in there is just verifying that UnsupportedOperationExceptions are thrown?

Reply:
Added tests for `AvroToProtoSerializer`, `BigQueryDefaultSink` and `BigQueryDefaultWriter`.